AvaloniaUI/Avalonia
# Add new constructor to AvaloniaDictionary and include unit tests #17311

#17311



## What does the pull request do?

This pull request adds a new constructor to `AvaloniaDictionary<TKey, TValue>` that accepts an existing `Dictionary<TKey, TValue>`. The goal is to provide an easier and more efficient way to initialize `AvaloniaDictionary` with pre-existing dictionary data.

## What is the current behavior?

Currently,` AvaloniaDictionary` lacks a constructor that directly accepts a `Dictionary<TKey, TValue>`. Developers have to manually copy elements into AvaloniaDictionary, which increases boilerplate code and is less efficient.

## What is the updated/expected behavior with this PR?

With this PR, developers can initialize AvaloniaDictionary with an existing dictionary directly, simplifying the code and reducing manual effort.
To test:

- You can create an AvaloniaDictionary by passing a populated dictionary and verify that all entries are correctly initialized.
- Passing null should throw an ArgumentNullException.

## Checklist


## Breaking changes

No breaking changes are introduced, as this PR simply adds an overload to the existing constructors.

## Obsoletions / Deprecations


## Fixed issues
